planning/obstacle_cruise_planner/src/node.cpp

```diff
@@ -641,6 +726,43 @@ std::vector<TargetObstacle> ObstacleCruisePlannerNode::filterObstacles(
   return target_obstacles;
 }
 
+void ObstacleCruisePlannerNode::updateHasStopped(std::vector<TargetObstacle> & target_obstacles)
+{
+  for (auto & obstacle : target_obstacles) {
+    const bool is_cruise_obstacle = isCruiseObstacle(obstacle.classification.label);
+    const bool is_stop_obstacle = isStopObstacle(obstacle.classification.label);
+
+    if (is_stop_obstacle && !is_cruise_obstacle) {
+      obstacle.has_stopped = true;
+      continue;
+    }
+
+    if (is_cruise_obstacle) {
+      const auto itr = std::find_if(
+        prev_target_obstacles_.begin(), prev_target_obstacles_.end(),
+        [&](const auto & prev_target_obstacle) {
+          return obstacle.uuid == prev_target_obstacle.uuid;
+        });
+      const bool has_already_stopped = (itr != prev_target_obstacles_.end()) && itr->has_stopped;
+      if (has_already_stopped) {
+        if (std::abs(obstacle.velocity) < obstacle_velocity_threshold_from_stop_to_cruise_) {
+          obstacle.has_stopped = true;
+          continue;
+        }
+      } else {
+        if (std::abs(obstacle.velocity) < obstacle_velocity_threshold_from_cruise_to_stop_) {
+          obstacle.has_stopped = true;
+          continue;
+        }
+      }
```

Comment:
Do we need 'if(has_already_stopped)' ? It seems we always do 'obstacle.has_stopeed=true' regardless of the value has_already stopped

Reply:
done